api/v1beta1/azuremachine_types.go

```diff
@@ -114,6 +114,8 @@ type AzureMachineSpec struct {
 	// SubnetName selects the Subnet where the VM will be placed
 	// +optional
 	SubnetName string `json:"subnetName,omitempty"`
+
+	NetworkInterfaces []AzureNetworkInterface `json:"networkInterfaces,omitempty"`
```

Comment:
Should this be an optional field? Just to clarify, if apply an existing template w/o the `NetworkInterfaces` field do we want it to give an error or just create an empty slice?

Reply:
Also we're going to want to add a header comment for this as well.